OneSignalSDK/unittest/src/test/java/com/test/onesignal/PushRegistratorHMSIntegrationTestsRunner.java

```diff
@@ -0,0 +1,128 @@
+package com.test.onesignal;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+
+import com.huawei.hms.common.ApiException;
+import com.huawei.hms.support.api.client.Status;
+import com.onesignal.InAppMessagingHelpers;
+import com.onesignal.OneSignal;
+import static com.onesignal.OneSignalPackagePrivateHelper.UserState.PUSH_STATUS_HMS_TOKEN_TIMEOUT;
+import static com.onesignal.OneSignalPackagePrivateHelper.UserState.PUSH_STATUS_HMS_API_EXCEPTION_OTHER;
+import com.onesignal.ShadowCustomTabsClient;
+import com.onesignal.ShadowCustomTabsSession;
+import com.onesignal.ShadowHmsInstanceId;
+import com.onesignal.ShadowOSUtils;
+import com.onesignal.ShadowOneSignalRestClient;
+import com.onesignal.ShadowPushRegistratorHMS;
+import com.onesignal.StaticResetHelper;
+import com.onesignal.example.BlankActivity;
+
+import org.json.JSONException;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.android.controller.ActivityController;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowLog;
+
+import static com.test.onesignal.RestClientAsserts.assertHuaweiPlayerCreateAtIndex;
+import static com.test.onesignal.RestClientAsserts.assertPlayerCreateNotSubscribedAtIndex;
+import static com.test.onesignal.RestClientAsserts.assertPlayerCreateSubscribedAtIndex;
+import static com.test.onesignal.RestClientAsserts.assertPlayerCreateWithNotificationTypesAtIndex;
+import static com.test.onesignal.RestClientAsserts.assertRestCalls;
+import static com.test.onesignal.TestHelpers.threadAndTaskWait;
+
+@Config(
+    packageName = "com.onesignal.example",
+    shadows = {
+        ShadowOSUtils.class,
+        ShadowOneSignalRestClient.class,
+        ShadowCustomTabsClient.class,
+        ShadowCustomTabsSession.class,
+        ShadowHmsInstanceId.class,
+        ShadowPushRegistratorHMS.class
+    },
+    sdk = 26
+)
+@RunWith(RobolectricTestRunner.class)
+public class PushRegistratorHMSIntegrationTestsRunner {
+
+    @SuppressLint("StaticFieldLeak")
+    private static Activity blankActivity;
+    private static ActivityController<BlankActivity> blankActivityController;
+
+    @BeforeClass // Runs only once, before any tests
+    public static void setUpClass() throws Exception {
+        ShadowLog.stream = System.out;
+        TestHelpers.beforeTestSuite();
+        StaticResetHelper.saveStaticValues();
+    }
+
+    @Before
+    public void beforeEachTest() throws Exception {
+        TestHelpers.beforeTestInitAndCleanup();
+
+        ShadowOSUtils.supportsHMS(true);
+
+        blankActivityController = Robolectric.buildActivity(BlankActivity.class).create();
+        blankActivity = blankActivityController.get();
+    }
+
+    private static void assertHuaweiSubscribe() throws JSONException {

```

Comment:
Should we throw these `Assert` helpers inside another `class` like `JsonAsserts` and `RestClientAsserts`
maybe `DeviceTypeAsserts`?

Reply:
It could go into `RestClientAsserts`, however it does include `assertRestCalls(2);` which is very specific to the UnitTest in this class.

A method code be added to `RestClientAsserts` that includes these two lines, but since this probably won't be used in any other test classes I think it best to leave as is.
```java
assertHuaweiPlayerCreateAtIndex(1);
assertPlayerCreateSubscribedAtIndex(1);
```